hazelcast/src/main/java/com/hazelcast/instance/impl/JetExtension.java

```diff
@@ -125,10 +128,10 @@ public void onClusterStateChange(ClusterState ignored) {
     }
 
     public void onClusterVersionChange(Version newVersion) {
-        if (!activated && newVersion.isGreaterOrEqual(Versions.V5_0)) {
+        if (!activated && isAfterStartCalled && newVersion.isGreaterOrEqual(Versions.V5_0)) {
```

Comment:
instead of introducing a volatile field, we can check the version of the member and if it is equal to the `newVersion` then it means we are not actually changing anything and just return. we will activate the jet in `afterStart` anyway.

```
        if (node.getVersion().asVersion().equals(newVersion)) {
            return;
        }
```

Reply:
We mainly use this method to activate Jet after rolling upgrade. If we introduce this early return, jet won't be enabled in the rolling upgrade scenarios.
Consider the scenario:
- 5.0 member joined 4.2 cluster.
- since cluster version < 5.0, jet won't be enabled first in `afterStart()` or initial `onClusterVersionChange`. 
- Then, let's say we're doing this cluster rolling upgrade. After the rolling upgrade, cluster version will be updated to 5.0, and we need to enable jet in this case. If we add this member version check there, we won't enable jet in this scenario since the member version and new cluster version would be same.

Reply:
I'm merging this PR because it's been open for a long time and I believe this discussion is resolved.